File: legacy/src/main/java/br/com/legacy/models/UserSuggestion.java
package br.com.legacy.models;

import android.os.Parcel;
import android.os.Parcelable;
import androidx.annotation.NonNull;

import com.linkedin.android.spyglass.mentions.Mentionable;

public class UserSuggestion implements Mentionable
{

    private final String name;
    private final String id;
    private String avatar = "";

    public UserSuggestion (String name, String id){
        this.name = name;
        this.id = id;
    }

    public UserSuggestion(String name, String id, String avatar) {
        this.name = name;
        this.id = id;
        this.avatar = avatar;
    }

    public String getName() {
        return name;
    }

    public String getId() {
        return id;
    }

    public String getAvatar() {
        return avatar;
    }

    @NonNull
    @Override
    public String getTextForDisplayMode(MentionDisplayMode mode) {
        return name;
    }

    @Override
    public MentionDeleteStyle getDeleteStyle() {
        return MentionDeleteStyle.PARTIAL_NAME_DELETE;
    }

    @Override
    public int getSuggestibleId() {
        return 0;
    }

    @Override
    public String getSuggestiblePrimaryText() {
        return name;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeString(id);
        dest.writeString(avatar);
    }

    public UserSuggestion(Parcel in){
        name = in.readString();
        id = in.readString();
        avatar = in.readString();
    }

    public static final Parcelable.Creator<UserSuggestion> CREATOR
            = new Parcelable.Creator<UserSuggestion>() {
        public UserSuggestion createFromParcel(Parcel in) {
            return new UserSuggestion(in);
        }

        public UserSuggestion[] newArray(int size) {
            return new UserSuggestion[size];
        }
    };
}

File: feature_feed/src/main/kotlin/br/com/goin/feature/feed/post/UserSuggestion.kt
package br.com.goin.feature.feed.post

import android.os.Parcel
import android.os.Parcelable
import androidx.annotation.NonNull
import com.linkedin.android.spyglass.mentions.Mentionable

class UserSuggestion : Mentionable {

    val name: String
    val id: String
    var avatar = ""
        private set

    constructor(name: String, id: String) {
        this.name = name
        this.id = id
    }

    constructor(name: String, id: String, avatar: String) {
        this.name = name
        this.id = id
        this.avatar = avatar
    }

    @NonNull
    override fun getTextForDisplayMode(mode: Mentionable.MentionDisplayMode): String {
        return name
    }

    override fun getDeleteStyle(): Mentionable.MentionDeleteStyle {
        return Mentionable.MentionDeleteStyle.PARTIAL_NAME_DELETE
    }

    override fun getSuggestibleId(): Int {
        return 0
    }

    override fun getSuggestiblePrimaryText(): String {
        return name
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeString(id)
        dest.writeString(avatar)
    }

    constructor(`in`: Parcel) {
        name = `in`.readString()!!
        id = `in`.readString()!!
        avatar = `in`.readString()!!
    }

    companion object CREATOR : Parcelable.Creator<UserSuggestion> {
        override fun createFromParcel(parcel: Parcel): UserSuggestion {
            return UserSuggestion(parcel)
        }

        override fun newArray(size: Int): Array<UserSuggestion?> {
            return arrayOfNulls(size)
        }
    }
}
